### ISSUE TITLE
[Accessibility] Accessibility skips or resets focus talkback in scrollable Column or ListView (Testing in Android)

### ISSUE THREAD

**Issue by rod-moraes** [flutter/flutter#173080]
### Steps to reproduce

Create a scrollable widget (e.g., SingleChildScrollView or ListView) containing a Column.

Add multiple children (e.g., Texts, Buttons) inside the Column.

Enable TalkBack on an Android device.

Try navigating through the screen using TalkBack gestures.

### Expected results

TalkBack should move focus linearly through each accessible item in order,, without skipping or looping.

### Actual results

TalkBack skips some items inside the scrollable widget.

In some cases, it jumps back to the top instead of progressing.

This happens more often when nesting widgets within the Column.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: _AnimationBackground(
          body: Semantics(
            explicitChildNodes: true,
            child: Column(
              mainAxisSize: MainAxisSize.min,
              children: <Widget>[
                ...List.generate(
                  100,
                  (index) => IndexedSemantics(
                    index: index,
                    child: Container(
                      padding: const EdgeInsets.all(80),
                      color: Colors.red,
                      child: Text('teste $index'),
                    ),
                  ),
                ),
              ],
            ),
          ),
        ),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

class _AnimationBackground extends StatelessWidget {
  final Widget body;
  const _AnimationBackground({required this.body});

  @override
  Widget build(BuildContext context) {
    return ScrollConfiguration(
      behavior: ScrollConfiguration.of(context).copyWith(scrollbars: false),
      child: ListView(children: [body]),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.32.5, on macOS 26.0 25A5316i darwin-arm64, locale pt-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 16.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.3)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report @rod-moraes 
I tried the given code sample with latest stable and master versions on Android as shown below and observed that I was able to traverse to each element properly upon scrolling up and down:

https://github.com/user-attachments/assets/8e57ea6f-d5e0-480b-964b-6f4b22ac011c

Please check if above is correct interpretation of behavior you are reporting. If not, please provide us a short video that shows the issue.

**Comment by rod-moraes** [comment]
When I swipe from left to right or right to left, it’s supposed to move to the next item. But as you can see in the video, at some point the focus gets lost — I can’t select anything anymore unless I perform a specific gesture.
However, for a blind user, this is a big problem because they lose all reference. The expected behavior is that the screen reader should continue moving to the next item without getting lost.

https://github.com/user-attachments/assets/4dd0d01e-e36a-466d-85a7-c2be83e1baff

**Comment by darshankawar** [comment]
Thanks for the update. Now I was able to replicate the reported behavior.

1. The focus moves to extra silent space between the elements (not the first item though) which maybe related / similar to https://github.com/flutter/flutter/issues/164483
2. The accessibility focus moves to the top of the screen after last visibile item on screen.


```
stable : 3.32.8
master : 3.33.0-1.0.pre-1255
```

**Comment by chunhtai** [comment]
not entirely sure what is going on, seems like an android embedding issue though

This looks like a series issue that may affect the usability quite a bit

**Comment by jmagman** [comment]
Possibly related to https://github.com/flutter/flutter/issues/144367#issuecomment-1992204114 ?

**Comment by jmagman** [comment]
@darshankawar can you confirm this _doesn't_ reproduce the same way with VoiceOver on iOS? That will help narrow it down.

**Comment by darshankawar** [comment]
> [@darshankawar](https://github.com/darshankawar) can you confirm this _doesn't_ reproduce the same way with VoiceOver on iOS? That will help narrow it down.

@jmagman Yes, it does seem to replicate on iOS as well, although the focus doesn't jump to first element or at the top unlike on Android:

https://github.com/user-attachments/assets/77d75fc6-f8a4-4cc7-8dd5-1cee1cc75fec


The focus isn't visible on last visible item on the screen although it does read the element properly.

**Comment by rod-moraes** [comment]
On iOS, when setting `cacheExtent: double.maxFinite,` it works.
But on Android, the issue still persists.

**Comment by ash2moon** [comment]
Hi @rod-moraes,

I've analyzed the code and it seems that the issue is related to using a `ListView` to allow scrolling on a single `Column` containing a long list of `Container` widgets. `ListView` is not meant to allow scrolling for nested objects; it's purpose is to allow scrolling for the immediate children of your widget. In this case, you only have one object. What's actually happening is your large Column is too tall and gets clipped when you reach a certain height. At this point, the ListView itself tries to go to the next item, but since there are no next items within the list, it moves the focus to the next item in the semantic widget tree.

I've verified that semantic scrolling works fine when using `SingleChildScrollView(child: body)` instead of `ListView(children: [body])`. It also works fine when we don't use a `Column` to display the large list of items but rather `ListView.builder`.

For long lists, it's highly recommended to use ListView.builder. It's much more performant because it only builds the items that are visible on screen. `ListView` also handle accessibility indexing automatically, so you don't need to manually wrap your widgets with IndexedSemantics as long as you keep the `addSemanticIndexes` property default to true.

About increasing `cacheExtent`: While that can be a valid workaround in some scrolling accessibility scenarios, it wouldn't apply here. In sample code, the ListView has only a single child (the Column), so there are no other list items for the accessibility focus to jump to, regardless of the cacheExtent.

**Comment by rod-moraes** [comment]
Hi @ash2moon @darshankawar, actually when I first brought up the issue, I was trying to reproduce a problem we’re seeing on one of our pages. Using that approach, I was able to replicate it — but the same problem also happens with a `CustomScrollView`.

Even when using all the semantics parts, for some reason it gets lost when there’s a very large spacing.

You can also notice that this only happens when there’s a pinned AppBar at the top.

Here’s an example:

```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: CustomScrollView(
          slivers: [
            SliverAppBar(title: Text(widget.title), pinned: true),
            SliverEnsureSemantics(
              sliver: SliverToBoxAdapter(
                child: Column(
                  mainAxisSize: MainAxisSize.min,
                  spacing: 200,
                  children: <Widget>[
                    ...List.generate(
                      100,
                      (index) => IndexedSemantics(
                        index: index,
                        child: Container(
                          padding: const EdgeInsets.all(80),
                          color: Colors.red,
                          child: Text('teste $index'),
                        ),
                      ),
                    ),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

**Comment by rod-moraes** [comment]
Same problem with `SingleChildScrollView`
```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(widget.title)),
      body: Center(
        child: SingleChildScrollView(
          child: Column(
            mainAxisSize: MainAxisSize.min,
            spacing: 200,
            children: <Widget>[
              ...List.generate(
                100,
                (index) => IndexedSemantics(
                  index: index,
                  child: Container(
                    padding: const EdgeInsets.all(80),
                    color: Colors.red,
                    child: Text('teste $index'),
                  ),
                ),
              ),
            ],
          ),
        ),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```

**Comment by ash2moon** [comment]
Can you please try `ListView.builder` and see if that works?

Also I just tried both of those code sample and neither of them seem to have an issue. Could you please provide your latest flutter doctor output and a sample video of what you're seeing with those code samples?

**Comment by rod-moraes** [comment]
@ash2moon 

Using the `CustomScrollView`, you can see that it depends on the spacing between the semantic items and their position on the screen. At the start of the video, you can see the issue happening, but when I scroll even slightly, it works. This happens because of the distance between the items—if the spacing were larger, the problem would occur more often.

Additionally, the problem is that I have lists and other components on the screen that cannot be managed with a ListView.builder, since they are separate components within a single screen. The example I gave is simple so you can reproduce the issue, but in the actual project I cannot replace the components. Because of that, this problem mainly occurs where I have large spacings depending on the context (for example, at the end of a background and the start of a bottom bar).

So I’d like to emphasize that this is not a case of “just use a ListView.builder,” because in this scenario it’s not possible to replace components. On this screen, I use a CustomScrollView with slivers, including sliver app bars, and the layout is responsive and depends on other elements.

https://github.com/user-attachments/assets/c537dc7a-8570-4cab-9f76-f127210a8dd3

**Comment by rod-moraes** [comment]
In this other example, it’s the example of the code I sent with `SingleChildScrollView`.

https://github.com/user-attachments/assets/627c987d-0481-45d5-a249-ae88ebf470f9

**Comment by ash2moon** [comment]
Hi @rod-moraes, the code samples I tried were from [this comment](https://github.com/flutter/flutter/issues/173080#issuecomment-3206403499) and [this comment](https://github.com/flutter/flutter/issues/173080#issuecomment-3206375337). Can you confirm that the videos you just posted are from those comments? If so, just to confirm can you list the device and `flutter doctor` output?

I was able to replicate the [issue from the original report](https://github.com/flutter/flutter/issues/173080#issue-3281717811), but the issue with that was due to ListView trying to render one large Column as I mentioned earlier.

**Comment by rod-moraes** [comment]
Hi @ash2moon 
Yes, my tests were with the code you mentioned. If it works fine for you, try increasing the spacing of the Column. I believe it has to do with this empty space causing things to get lost.

Adb devices: `RQ8R9035DLV            device 1-0 product:m62xx model:SM_M625F device:m62 transport_id:5` 

Flutter Doctor:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.32.5, on macOS 26.0 25A5316i darwin-arm64, locale pt-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 16.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.3)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
```
